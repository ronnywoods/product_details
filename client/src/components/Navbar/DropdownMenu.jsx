import React from 'react';

const DropdownMenu = props => {
  return (
    <div className="col-5">
      <ul>
        <li className="headline">{props.title}</li>
        {props.options.map((op,i ) => {
          if(op === '____'){
            return (
              <li key = {op+i} className="gray">{op}</li>
            )
          } else if (op === 'Sale'){
            return (
              <li key = {op+i} className="bold">{op}</li>
            )
          } else {
            return (
              <li key={op+i}>{op}</li>
            )
          }
        })}
      </ul>
    </div>
  )
};

export default DropdownMenu;
