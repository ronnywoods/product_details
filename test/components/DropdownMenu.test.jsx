import React from 'react';
import { shallow } from 'enzyme';

import DropdownMenu from '../../client/src/components/Navbar/DropdownMenu';
import menu from '../../client/src/components/Navbar/data';

describe('Dropdown Menu Component', () => {
  const component = shallow(<DropdownMenu title='FEATURED' options={menu.women.featured} />);

  it('should match snapshot when props are passed in', () => {
    expect(component).toMatchSnapshot();
  });

  it('should contain only 1 headline', () => {
    expect(component.find('.headline')).toHaveLength(1);
  });

  it('should have more than 3 option', () => {
    expect(component.find('li').length).toBeGreaterThan(3);
  });
});
