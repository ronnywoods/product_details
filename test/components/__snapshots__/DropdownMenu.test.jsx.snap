// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Menu Component should match snapshot when props are passed in 1`] = `
<div
  className="col-5"
>
  <ul>
    <li
      className="headline"
    >
      FEATURED
    </li>
    <li
      key="New Arrivals0"
    >
      New Arrivals
    </li>
    <li
      key="Best Sellers1"
    >
      Best Sellers
    </li>
    <li
      key="Release Dates2"
    >
      Release Dates
    </li>
    <li
      className="gray"
      key="____3"
    >
      ____
    </li>
    <li
      key="YEEZY4"
    >
      YEEZY
    </li>
    <li
      key="Ultraboost5"
    >
      Ultraboost
    </li>
    <li
      key="NMD6"
    >
      NMD
    </li>
    <li
      key="Statement Sneakers7"
    >
      Statement Sneakers
    </li>
    <li
      key="Workout Essentials8"
    >
      Workout Essentials
    </li>
    <li
      className="gray"
      key="____9"
    >
      ____
    </li>
    <li
      className="bold"
      key="Sale10"
    >
      Sale
    </li>
  </ul>
</div>
`;
